src/main/java/hudson/plugins/analysis/core/AnnotationDifferencer.java

```diff
@@ -40,7 +40,7 @@
     private static Set<FileAnnotation> difference(
             final Set<FileAnnotation> target, final Set<FileAnnotation> other) {
         Set<FileAnnotation> difference = Sets.newHashSet(target);
-        difference.removeAll(other);
+        difference.removeAll(Sets.newHashSet(other));
         return difference;
```

Comment:
I haven't tested performance of this method yet -- just the performance of removeAll separately. Do you use any library for benchmarking? If not, I can try to find time to create a java program that tests the performance of this method with different sets and set sizes.